backend/cellguide/pipeline/computational_marker_genes/computational_markers.py

```diff
@@ -238,11 +162,122 @@ def _prepare_cell_counts_and_gene_expression_dfs(
         # remake the multi-index
         universe_cell_counts_df = universe_cell_counts_df.groupby(self.groupby_terms_with_celltype).sum()
-        return universe_cell_counts_df, expressions_df
 
-    def get_computational_marker_genes(self) -> dict[str, list[ComputationalMarkerGenes]]:
+        # create an unrolled copy of the cell counts dataframe
+        cell_counts_df_orig = universe_cell_counts_df.copy()
+        cell_counts_df_orig["n_cells"] = 0
+        cell_counts_df_orig["n_cells"][cell_counts_df.index] = cell_counts_df["n_cells"]
+        return universe_cell_counts_df, cell_counts_df_orig, expressions_df
+
+    def _process_cell_type__parallel(
+        self,
+        *,
+        i: int,
+        cell_types_o: np.ndarray,
+        e_sum_o: np.ndarray,
+        e_sqsum_o: np.ndarray,
+        n_cells_o: np.ndarray,
+        e_sum_o_orig: np.ndarray,
+        e_sqsum_o_orig: np.ndarray,
+        n_cells_o_orig: np.ndarray,
+        filter_genes: np.ndarray,
+    ) -> tuple[str, np.ndarray, np.ndarray, np.ndarray]:
+        """
+        This function is used in dask scheduler to process each cell type in parallel.
+
+        Parameters
+        ----------
+        i : int
+            The index of the current cell type being processed.
+        cell_types_o : np.ndarray
+            Array of cell types.
+        e_sum_o : np.ndarray
+            Array of gene expression sums for each cell type.
+        e_sqsum_o : np.ndarray
+            Array of squared gene expression sums for each cell type.
+        n_cells_o : np.ndarray
+            Array of cell counts for each cell type.
+        e_sum_o_orig : np.ndarray
+            Array of raw gene expression sums for each cell type.
+        e_sqsum_o_orig : np.ndarray
+            Array of raw squared gene expression sums for each cell type.
+        n_cells_o_orig : np.ndarray
+            Array of cell counts for each cell type.
+        filter_genes: np.ndarray
+            A boolean array indicating which genes to filter out.
+
+        Returns
+        -------
+        tuple[np.ndarray, np.ndarray]
+            A tuple containing the filtered effect sizes and the column indices of the remaining genes.
+        """
+        e_sum_o = e_sum_o.copy()
+        e_sqsum_o = e_sqsum_o.copy()
+        n_cells_o = n_cells_o.copy()
+        cell_type_target = cell_types_o[i]
+
+        indexer = pd.Series(index=cell_types_o, data=np.arange(cell_types_o.size))
+
+        for j, cell_type in enumerate(cell_types_o):
+            if cell_type_target == cell_type or are_cell_types_colinear(cell_type, cell_type_target):
+                continue
+
+            overlapping_descendants = get_overlapping_cell_type_descendants(cell_type, cell_type_target)
+            overlapping_descendants = list(set(overlapping_descendants).intersection(cell_types_o))
+            if len(overlapping_descendants) > 0:
```

Comment:
Subtract out expresison and cell count metrics of overlapping descendants

Reply:
Nit: does this also affect how we're calculating N cells expressing a given gene when we do the final t-test?